src/language-js/print/object.js

```diff
@@ -118,7 +118,8 @@ function printObject(path, options, print) {
     if (
       (prop.node.type === "TSPropertySignature" ||
         prop.node.type === "TSMethodSignature" ||
-        prop.node.type === "TSConstructSignatureDeclaration") &&
+        prop.node.type === "TSConstructSignatureDeclaration" ||
+        prop.node.type === "TSCallSignatureDeclaration") &&
```

Comment:
Looks like these type are not like `Property`(semi belong to them, but comma not belong to `Property`).

Have to say we are not printing the correctly, but not your fault. We should fix in future.